http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -142,14 +143,41 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         )
         .run(authContext)
 
+      // Poll app status
+      appOk <- pollAppsForSuccess(landingZoneResources.relayNamespace)
+
+      _ <-
+        if (appOk && app.status == AppStatus.Running)
+          F.unit
+        else
+          F.raiseError[Unit](
+            AppCreationException(
+              s"App ${params.appName.value} failed to start in cluster ${landingZoneResources.clusterName.value} in cloud context ${params.cloudContext.asString}",
+              Some(ctx.traceId)
+            )
+          )
+
+
       _ <- logger.info(ctx.loggingCtx)(
         s"Finished app creation for app ${params.appName.value} in cluster ${landingZoneResources.clusterName.value} in cloud context ${params.cloudContext.asString}"
       )
 
-      // TODO (TOAZ-229): poll app for completion
-
     } yield ()
 
+  private def pollAppsForSuccess(relayNamespace: RelayNamespace)(implicit ev: Ask[F, AppContext]): F[Boolean] = {
+    val relayBaseUri = Uri.unsafeFromString(s"https://${relayNamespace.value}.servicebus.windows.net")
+    val cromwellStatusUri = relayBaseUri / "cromwell" / "api" / "engine" / "v1" / "status"
+    for {
+
+      headers <- samDao.getLeoAuthToken.map(x => Headers(x))
```

Comment:
I will test but not sure the Leo auth token will work here since it's not a member of the `kubernetes-app` Sam resource.

We may want to use the UAMI token instead (which we also need for calling the LZ API I think)… I can look into this.